src/realm/util/encrypted_file_mapping.cpp

```diff
@@ -585,47 +585,62 @@ void EncryptedFileMapping::reclaim_page(size_t page_ndx)
  * 3) A scanning of 1K entries in the "don't scan" array (corresponding to 4M pages)
  * Approximately
  */
-size_t EncryptedFileMapping::reclaim_untouched(size_t& progress_index, size_t& work_limit) noexcept
+void EncryptedFileMapping::reclaim_untouched(size_t& progress_index, size_t& work_limit) noexcept
 {
 	const auto scan_amount_per_workunit = 4096;
-	auto done_some_work = [&]() { if (work_limit > 0) work_limit--; };
-
-    const size_t num_pages = m_page_state.size();
-    if (progress_index < m_first_page)
-        return 0;
-    if (progress_index >= get_end())
-        return 0;
-
-    size_t num_reclaimed = 0;
-    size_t next_scan_payment = scan_amount_per_workunit;
     size_t contiguous_scan_count = 0;
```

Comment:
Can't this just be a flag named `contiguous_scan` that is initially true, but set to false if a page is reclaimed?

Reply:
It could, but as far as I can see it will end up more convoluted.